### PR TITLE
Trace Kubernetes API calls using OpenTelemetry

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [x] New feature (non-breaking change which adds functionality)

**What is the new behavior (if this is a feature change)?**

OpenTelemetry offers instrumentation packages that are agnostic to the tool(s) that finally store telemetry data. When the SDK for Go reaches GA, we can leverage the OTLP exporter to offload further configuration to an external OpenTelemetry Collector.

For now, traces can be sent to one of a handful of destinations chosen by environment variable: Jaeger agent, Jaeger collector, stdout as JSON, or a file as JSON.

**Other information**:

Issue: [ch9735]

### CODE REVIEW

**Comment by cbandy** [comment]
Added some comments about OpenTelemetry configuration and the default tracer.

File: cmd/postgres-operator/main.go [hunk 1]
```diff
@@ -39,6 +39,12 @@ import (
 )
 
 func main() {
+	if flush, err := initOpenTelemetry(); err != nil {
+		log.Error(err)
```
Comment (jkatz) [inline cmd/postgres-operator/main.go:43]: Perhaps `log.Fatal(err)`?

File: cmd/postgres-operator/main.go [hunk 2]
```diff
@@ -53,7 +59,18 @@ func main() {
 	//give time for pgo-event to start up
 	time.Sleep(time.Duration(5) * time.Second)
```
Comment (jkatz) [inline cmd/postgres-operator/main.go:60]: ...since we're here, can we remove this?

**Review by jkatz — APPROVED** [review]
A few minor things, if that. Looks good.

**Pull request merged**